[Bug]: panic when using rsbuild

### System Info

```

  System:
    OS: Linux 7.0 cpe:/o:nixos:nixos:26.05 26.05 (Yarara)
    CPU: (20) x64 13th Gen Intel(R) Core(TM) i5-13600KF
    Memory: 39.79 GB / 62.62 GB
    Container: Yes
    Shell: 5.9 - /run/current-system/sw/bin/zsh
  Binaries:
    Node: 25.9.0 - /etc/profiles/per-user/meyer/bin/node
    npm: 11.12.1 - /etc/profiles/per-user/meyer/bin/npm
    pnpm: 10.28.2 - /etc/profiles/per-user/meyer/bin/pnpm
    bun: 1.3.11 - /etc/profiles/per-user/meyer/bin/bun
    Deno: 2.7.12 - /etc/profiles/per-user/meyer/bin/deno
  Browsers:
    Brave Browser: 147.1.89.132
    Chromium: 147.0.7727.101
    Firefox: 149.0.2
    Firefox Developer Edition: 149.0.2
  npmPackages:
    @rspack/core: ^1.7.5 => 1.7.5 

```

### Details

```
❯ RUST_BACKTRACE=full pnpm rsbuild build
Rsbuild v2.0.3

info    build started...
error   build failed in 3.39 s (client)
Panic occurred at runtime. Please file an issue on GitHub with the panic info and backtrace below: https://github.com/web-infra-dev/rspack/issues

thread 'tokio-8' (175890) panicked at crates/rspack_plugin_library/src/module_library_plugin.rs:106:8:
name can't be empty
stack backtrace:
Panic occurred at runtime. Please file an issue on GitHub with the panic info and backtrace below: https://github.com/web-infra-dev/rspack/issues
   0:     0x7fed66d8cfc0 - <unknown>

thread 'tokio-4' (175886) panicked at crates/rspack_plugin_library/src/module_library_plugin.rs:106:8:
name can't be empty
stack backtrace:
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command was killed with SIGABRT (Aborted): rsbuild build
```

### Reproduce link

_No response_

### Reproduce Steps

~~This issue is consistent; however, I do not know what causes it. I am only opening an issue because that was requested in the panic message.~~

https://github.com/web-infra-dev/rspack/issues/13918#issuecomment-4375613065

--- a/crates/rspack_plugin_library/src/module_library_plugin.rs
+++ b/crates/rspack_plugin_library/src/module_library_plugin.rs
@@ -101,9 +101,6 @@ async fn render_startup(
       .chunk_by_ukey
       .expect_get(chunk_ukey);
     let info_name = export_info.name().expect("should have name");
-    let used_name = export_info
-      .get_used_name(Some(info_name), Some(chunk.runtime()))
-      .expect("name can't be empty");
     let var_name = format!("{exports_name}{}", to_identifier(info_name));
 
     if info_name == "default"
@@ -116,6 +113,11 @@ async fn render_startup(
         "var {var_name} = {exports_name};\n",
       )));
     } else {
+      // Skip exports unused in this runtime (matches webpack's `if (!exportInfo.provided) continue;`).
+      let Some(used_name) = export_info.get_used_name(Some(info_name), Some(chunk.runtime()))
+      else {
+        continue;
+      };
       source.add(RawStringSource::from(format!(
         "var {var_name} = {};\n",
         match used_name {

